### PR TITLE
HIVE-27430: Upgrade Antlr in hplsql to 4.9.3

### PR DESCRIPTION
**Pull request by kasakrisz**
### What changes were proposed in this pull request?
Upgrade antlr to 4.9.3

### Why are the changes needed?
Synch with HMS

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
```
mvn test -Dtest=TestHplsqlLocal -pl hplsql
mvn test -Dtest=TestHplSqlViaBeeLine#testUdfDecimal -pl itests/hive-unit -Pitests -Drat.skip -am
```

### CODE REVIEW

**Review by mdayakar — COMMENTED** [review]
You may need to update the '**_antlr4-runtime_**' version in below files also, please check.

https://github.com/apache/hive/blob/master/bin/ext/hplsql.sh#L24
https://github.com/apache/hive/blob/master/hplsql/src/main/java/org/apache/hive/hplsql/Exec.java#L644

**Review by mdayakar — APPROVED** [review]

File: hplsql/src/main/java/org/apache/hive/hplsql/Exec.java [hunk 1]
```diff
@@ -662,6 +660,31 @@ public void registerUdf() {
...
+
+    for(String jarName: files) {
+      if(jarName.startsWith(prefix) && jarName.endsWith(".jar")) {
+        return jarName;
+      }
+    }
+
+    LOG.warn("No jar file found in directory '{}' with prefix '{}'", dir, prefix);
+    return null;
+  }
```
Comment (veghlaci05) [inline hplsql/src/main/java/org/apache/hive/hplsql/Exec.java:686]: You should use `File.list(FilenameFilter filter)` instead

File: hplsql/src/main/java/org/apache/hive/hplsql/Exec.java [hunk 2]
```diff
@@ -662,6 +660,31 @@ public void registerUdf() {
     udfRegistered = true;
   }
 
+  private static void addJar(ArrayList<String> sql, String dir, String jarName) {
+    if (isNotBlank(jarName)) {
+      sql.add("ADD JAR " + Paths.get(dir, jarName));
+    }
+  }
+
+  private static final Logger LOG = LoggerFactory.getLogger(Exec.class);
```
Comment (veghlaci05) [inline hplsql/src/main/java/org/apache/hive/hplsql/Exec.java:669]: I think this should be declared at the top of the class.

File: hplsql/src/main/java/org/apache/hive/hplsql/Exec.java [hunk 3]
```diff
@@ -662,6 +660,31 @@ public void registerUdf() {
     udfRegistered = true;
   }
 
+  private static void addJar(ArrayList<String> sql, String dir, String jarName) {
+    if (isNotBlank(jarName)) {
+      sql.add("ADD JAR " + Paths.get(dir, jarName));
+    }
+  }
```
Comment (veghlaci05) [inline hplsql/src/main/java/org/apache/hive/hplsql/Exec.java:667]: You may call `findJarLike` from this method. For me it's a bit odd to pass null value from outside and check it here.

File: hplsql/src/main/java/org/apache/hive/hplsql/Exec.java (hunk 1 above)
Comment (kasakrisz) [reply hplsql/src/main/java/org/apache/hive/hplsql/Exec.java]: done

File: hplsql/src/main/java/org/apache/hive/hplsql/Exec.java (hunk 3 above)
Comment (kasakrisz) [reply hplsql/src/main/java/org/apache/hive/hplsql/Exec.java]: done

**Review by veghlaci05 — APPROVED** [review]
LGTM, pending tests

**Pull request merged**